service/src/main/java/bio/terra/axonserver/app/controller/CromwellWorkflowController.java

```diff
@@ -0,0 +1,144 @@
+              .status(workflowStatus.getStatus()),
+          HttpStatus.OK);
+    } catch (bio.terra.cromwell.client.ApiException e) {
+      throw new ApiException("Error %s: %s".formatted(e.getCode(), e.getResponseBody()));
+    }
+  }
+
+  @Override
+  public ResponseEntity<ApiWorkflowIdAndLabel> getWorkflowLabels(
+      UUID workspaceId, UUID workflowId) {
+    try {
+      cromwellWorkflowService.validateWorkspaceAccessAndWorkflowLabelMatches(
+          workflowId, workspaceId, getToken().getToken());
+      CromwellApiLabelsResponse workflowLabels = cromwellWorkflowService.getLabels(workflowId);
+      return new ResponseEntity<>(
+          new ApiWorkflowIdAndLabel()
+              .id(UUID.fromString(workflowLabels.getId()))
+              .labels(workflowLabels.getLabels()),
+          HttpStatus.OK);
+    } catch (bio.terra.cromwell.client.ApiException e) {
+      throw new ApiException("Error %s: %s".formatted(e.getCode(), e.getResponseBody()));
+    }
+  }
+
+  @Override
+  public ResponseEntity<ApiWorkflowMetadataResponse> getWorkflowMetadata(
+      UUID workspaceId,
+      UUID workflowId,
+      @Nullable List<String> includeKey,
+      @Nullable List<String> excludeKey,
+      @Nullable Boolean expandSubWorkflows) {
+    try {
+      cromwellWorkflowService.validateWorkspaceAccessAndWorkflowLabelMatches(
+          workflowId, workspaceId, getToken().getToken());
+      CromwellApiWorkflowMetadataResponse workflowMetadata =
+          cromwellWorkflowService.getMetadata(
+              workflowId, includeKey, excludeKey, expandSubWorkflows);
+      return new ResponseEntity<>(
+          cromwellWorkflowService.toApiMetadataResponse(workflowMetadata), HttpStatus.OK);
+    } catch (bio.terra.cromwell.client.ApiException e) {
+      throw new ApiException("Error %s: %s".formatted(e.getCode(), e.getResponseBody()));
+    }
+  }
+
+  @Override
+  public ResponseEntity<ApiWorkflowQueryResponse> getWorkflowQuery(
+      UUID workspaceId,
+      @Nullable @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) Date submission,
+      @Nullable @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) Date start,
+      @Nullable @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) Date end,
+      @Nullable List<String> status,
+      @Nullable List<String> name,
+      @Nullable List<String> id,
+      @Nullable List<String> label,
+      @Nullable List<String> labelor,
+      @Nullable List<String> excludeLabelAnd,
+      @Nullable List<String> excludeLabelOr,
+      @Nullable List<String> additionalQueryResultFields,
+      @Nullable Boolean includeSubworkflows) {
+    // Check if the user has access to the workspace.
+    wsmService.checkWorkspaceReadAccess(workspaceId, getToken().getToken());
+    try {
+      if (label == null) {
+        label = new ArrayList<>();
+      }
+      // Restrict the subset to only workflows with the corresponding workspace id label.
+      label.add("%s:%s".formatted(WORKSPACE_ID_LABEL_KEY, workspaceId));
```

Comment:
we need to ensure noone can add some kind of "label or condition" to query other workspaces.

Suggest we filter out any user provided params that query for things like WORKSPACE_ID_LABEL_KEY, for example.

Reply:
I actually wonder if to make this simple we should just remove all of the label parameters in this request. I don't think we need to support those for querying. IE don't support label, labelor, excludeLabelAnd, excludeLabelOr